webpack.config.js

```diff
@@ -0,0 +1,109 @@
+const webpack = require("webpack");
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const ServiceWorkerWebpackPlugin = require("serviceworker-webpack-plugin");
+
+const config = {
+  entry: {
+    main: ["babel-polyfill", "./js/MainPage/index.js"]
+  },
+  output: {
+    path: path.resolve(__dirname, "dist"),
+    filename: "[name].bundle[hash].js"
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        use: "babel-loader",
+        exclude: /node_modules/
+      },
+      {
+        test: /\.css$/,
+        use: [
+          "style-loader",
+          {
+            loader: "css-loader",
+            options: {
+              importLoaders: 1
+            }
+          },
+          "postcss-loader"
+        ]
+      },
+      {
+        test: /\.(woff(2)?|ttf|eot)(\?v=\d+\.\d+\.\d+)?$/,
+        use: [
+          {
+            loader: "file-loader",
+            options: {
+              name: "[name].[ext]",
+              outputPath: "fonts/"
+            }
+          }
+        ]
+      },
+      {
+        test: /opencascade\.wasm\.wasm$/,
+        type: "javascript/auto",
+        loader: "file-loader"
+      }
+    ]
+  },
+  node: {
+    fs: "empty"
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: path.resolve(__dirname, "index.html")
+    }),
+    new ServiceWorkerWebpackPlugin({
+      entry: path.join(__dirname, 'service-worker.js'),
+    }),
+    new MonacoWebpackPlugin(),
+    new CopyPlugin({
+      patterns: [
+        {
+          from: "js/StandardLibraryIntellisense.ts",
+          to: "js/StandardLibraryIntellisense.ts"
+        },
+        {
+          from: "static_node_modules/opencascade.js/dist/oc.d.ts",
+          to: "opencascade.d.ts"
+        },
+        {
+          from: "node_modules/three/src/Three.d.ts",
+          to: "Three.d.ts"
```

Comment:
Making these three definition files available as assets for the web-app to grab later, an alternative to this would be another rule that matches on these specific files to import them as text (instead of as code) so that they would be added to the main bundle instead of needing to be fetched later.